EIPS/eip-225.md

```diff
@@ -0,0 +1,474 @@
+
+## Specification
+
+We define the following constants:
+
+ * **`EPOCH_LENGTH`**: Number of blocks after which to checkpoint and reset the pending votes.
+   * Suggested `30000` for the testnet to remain analogous to the mainnet `ethash` epoch.
+ * **`BLOCK_PERIOD`**: Minimum difference between two consecutive block's timestamps.
+   * Suggested `15s` for the testnet to remain analogous to the mainnet `ethash` target.
+ * **`EXTRA_VANITY`**: Fixed number of extra-data prefix bytes reserved for signer *vanity*.
+   * Suggested `32 bytes` to retain the current extra-data allowance and/or use.
+ * **`EXTRA_SEAL`**: Fixed number of extra-data suffix bytes reserved for signer seal.
+   * `65 bytes` fixed as signatures are based on the standard `secp256k1` curve.
+ * **`NONCE_AUTH`**: Magic nonce number `0xffffffffffffffff` to vote on adding a new signer.
+ * **`NONCE_DROP`**: Magic nonce number `0x0000000000000000` to vote on removing a signer.
+ * **`UNCLE_HASH`**: Always `Keccak256(RLP([]))` as uncles are meaningless outside of PoW.
+ * **`DIFF_NOTURN`**: Block score (difficulty) for blocks containing out-of-turn signatures.
+   * Suggested `1` since it just needs to be an arbitrary baseline constant.
+ * **`DIFF_INTURN`**: Block score (difficulty) for blocks containing in-turn signatures.
+   * Suggested `2` to show a slight preference over out-of-turn signatures.
+
+We also define the following per-block constants:
+
+ * **`BLOCK_NUMBER`**: Block height in the chain, where the height of the genesis is block `0`.
+ * **`SIGNER_COUNT`**: Number of authorized signers valid at a particular instance in the chain.
+ * **`SIGNER_INDEX`**: Index of the block signer in the sorted list of current authorized signers.
+ * **`SIGNER_LIMIT`**: Number of consecutive blocks out of which a signer may only sign one.
+   * Must be `floor(SIGNER_COUNT / 2) + 1` to enforce majority consensus on a chain.
+
+We repurpose the `ethash` header fields as follows:
+
+ * **`beneficiary`**: Address to propose modifying the list of authorized signers with.
+   * Should be filled with zeroes normally, modified only while voting.
+   * Arbitrary values are permitted nonetheless (even meaningless ones such as voting out non signers) to avoid extra complexity in implementations around voting mechanics.
+   * **Must** be filled with zeroes on checkpoint (i.e. epoch transition) blocks.
+ * **`nonce`**: Signer proposal regarding the account defined by the `beneficiary` field.
+   * Should be **`NONCE_DROP`** to propose deauthorizing `beneficiary` as a existing signer.
```

Comment:
From discussions on Pantheon github - should the NONCE_DROP be defined as the default value of the nonce field when not voting?

Reply:
well that is ture already.